Readability.js

```diff
@@ -624,6 +624,11 @@ Readability.prototype = {
 
     var pageCacheHtml = page.innerHTML;
 
+    var articleTags = this._doc.getElementsByTagName("article");
+    if (articleTags.length === 1) {
+      this._doc.body.innerHTML = articleTags[0].parentNode.innerHTML;
```

Comment:
>> This has a performance penalty: we'll be reparsing the HTML. I would suggest:
`this._doc.body.parentNode.replaceChild(articleTags[0].parentNode, this._doc.body)` which is almost O(1).

> Replacing the `<body>` tag with an `<article>` one? Anyway, that breaks everything.

Ah, d'oh. It's clear I didn't think this through. In that case, I guess:

`this._doc.body.innerHTML = ''; this._doc.body.appendChild(articleTags[0]);`

That or using `_setNodeTag` to make it a `<body>` again.